src/sign/interactive_sign/participant.rs

```diff
@@ -82,14 +153,31 @@ impl ProtocolParticipant for InteractiveSignParticipant {
         todo!()
     }
 
-    #[allow(unused)]
     fn new(
         sid: Identifier,
         id: ParticipantIdentifier,
         other_participant_ids: Vec<ParticipantIdentifier>,
         input: Self::Input,
     ) -> Result<Self> {
-        todo!()
+        let Input {
+            message_digest,
+            presign_input,
+        } = input;
+
+        let signing_material = SigningMaterial::PartialInput {
```

Comment:
I think we should make a constructor for `SigningMaterial` so this becomes a fn call

Reply:
Added, but not sure it's quite what you were thinking.